python/seldon_core/seldon_client.py

```diff
@@ -1426,19 +1448,22 @@ def rest_predict_gateway(
     metaKV = prediction_pb2.Meta()
     metaJson = {"tags": meta}
     json_format.ParseDict(metaJson, metaKV)
-
-    if bin_data is not None:
-        request = prediction_pb2.SeldonMessage(binData=bin_data, meta=metaKV)
-    elif str_data is not None:
-        request = prediction_pb2.SeldonMessage(strData=str_data, meta=metaKV)
-    elif json_data is not None:
-        request = json_to_seldon_message({"jsonData": json_data})
+    if raw_data is not None:
+        request = json_to_seldon_message(raw_data)
```

Comment:
Is the `request` variable needed here? It seems that after the big `if` / `else` only `payload` is used.

Reply:
used also in line #1556 (return SeldonClientPrediction(request, response, success, msg))